kotlinpoet-metadata-specs/src/main/kotlin/com/squareup/kotlinpoet/metadata/specs/internal/KmTypes.kt

```diff
@@ -148,14 +148,7 @@ internal fun ImmutableKmType.toTypeName(
 internal fun ImmutableKmTypeParameter.toTypeVariableName(
   typeParamResolver: ((index: Int) -> TypeName)
 ): TypeVariableName {
-  val finalVariance = variance.toKModifier().let {
-    if (it == KModifier.OUT) {
-      // We don't redeclare out variance here
-      null
-    } else {
-      it
-    }
-  }
+  val finalVariance = variance.toKModifier()
```

Comment:
This was a bug revealed by the regression test I added, opportunistically fixed here. Was a hold-over from Moshi's handling of metadata